server/src/test/java/org/opensearch/transport/TransportServiceHandshakeTests.java

```diff
@@ -223,6 +226,34 @@ public void testNodeConnectWithDifferentNodeId() {
         assertFalse(handleA.transportService.nodeConnected(discoveryNode));
     }
 
+    public void testNodeConnectWithDifferentNodeIDSkipValidation() throws IllegalAccessException {
+        Settings settings = Settings.builder().put("cluster.name", "test").build();
+        NetworkHandle handleA = startServices("TS_A", settings, Version.CURRENT);
+        NetworkHandle handleB = startServices("TS_B", settings, Version.CURRENT);
+        DiscoveryNode discoveryNode = new DiscoveryNode(
+            randomAlphaOfLength(10),
+            handleB.discoveryNode.getAddress(),
+            emptyMap(),
+            emptySet(),
+            handleB.discoveryNode.getVersion()
+        );
+        try (MockLogAppender mockLogAppender = MockLogAppender.createForLoggers(LogManager.getLogger(TransportService.class))) {
+
+            mockLogAppender.addExpectation(
+                new MockLogAppender.SeenEventExpectation(
+                    "Validation Skipped",
+                    "org.opensearch.transport.TransportService",
+                    Level.INFO,
+                    "Connection validation was skipped"
+                )
+            );
+
+            handleA.transportService.connectToNode(discoveryNode, TestProfiles.LIGHT_PROFILE, true);
+
+            mockLogAppender.assertAllExpectationsMatched();
```

Comment:
Let's test this feature using assert statements. Something similar [like](https://github.com/opensearch-project/OpenSearch/blob/ee626f11b8a1524ef3ce1cc43b6aa226173f45e5/server/src/test/java/org/opensearch/transport/TransportServiceHandshakeTests.java#L210)